kbase-extension/static/kbase/js/common/cellComponents/tabs/jobStatus/jobStateList.js

```diff
@@ -67,91 +77,166 @@ define([
-                $container.append($(createTable()));
+        function startJobListener(jobId) {
+            listeners.push(
+                bus.listen({
+                    channel: {
+                        jobId: jobId,
+                    },
+                    key: {
+                        type: 'job-status',
+                    },
+                    handle: handleJobStatusUpdate,
+                })
+            );
+        }
+
+        /**
+         * parse and update the row with job info
+         * @param {Object} message
+         */
+        function handleJobInfoUpdate(message) {
+            if (message.jobId && message.jobInfo && widgetsById[message.jobId]) {
+                widgetsById[message.jobId].updateParams(message.jobInfo);
+            }
+        }
 
-                const numberOfChildJobs = arg.childJobs.length;
+        /**
+         * Pass the job state to all row widgets
+         * @param {Object} message
+         */
+        function handleJobStatusUpdate(message) {
+            const jobState = Jobs.updateJobModel(message.jobState);
 
-                return Promise.try(() => {
-                    arg.childJobs.forEach((childJob, index) => {
-                        createJobStateWidget(index, childJob.job_id, childJob.status);
-                    });
-                }).then(() => {
-                    renderTable($container, numberOfChildJobs);
+            if (jobState.child_jobs) {
+                jobState.child_jobs.forEach((state) => {
+                    widgetsById[state.job_id].updateState(state);
                 });
-            });
+            }
+
+            if (jobState.job_id && widgetsById[jobState.job_id]) {
+                widgetsById[jobState.job_id].updateState(jobState);
+            }
         }
 
         /**
-         * Creates a job state widget and puts it in the widgets object, keyed by the index.
-         * This assumes that all child job states will always be returned in the same order
-         * on each state lookup, with new ones added to the end of the list. This also adds
-         * to the bottom of the job state list table.
+         * Creates a job state widget and puts it in the widgetsById object, keyed by the jobId.
          *
-         * Each job state widget knows which child job index its in, so it can look up its
-         * state as well.
-         * @param {int} jobIndex
-         * @param {string} jobId
+         * @param {object} jobState  -- jobState object, containing jobID, status, etc.
+         * @param {int}    jobIndex
          */
-        function createJobStateWidget(jobIndex, jobId, initialState) {
+
+        function createJobStateListRowWidget(jobState, jobIndex) {
+            widgetsById[jobState.job_id] = JobStateListRow.make();
             return Promise.try(() => {
-                widgets[jobIndex] = JobStateListRow.make({
-                    model: model,
+                widgetsById[jobState.job_id].start({
```

Comment:
This doesn't need to be wrapped in a `Promise.try`, `JobStateListRow.start` is already a promise.